backends/ebpf/README.md

```diff
@@ -335,4 +335,75 @@ The following tests run ebpf programs:
    Note: by default the kernel ebpf tests are disabled; if you want to enable them
    you can modify the file `backends/ebpf/CMakeLists.txt` by setting this variable to `True`:
    `set (SUPPORTS_KERNEL True)`
-   
\ No newline at end of file
+   
+# How to inject custom extern function to the generated eBPF program?
+
+The P4 to eBPF compiler comes with the support for custom C extern functions. It means that a developer
+can write a custom, eBPF-compatible (acceptable by BPF verifier) C function and call it
+from the P4 program as a normal P4 action. As a result, P4 program can be extended with
+functionality, which is not supported natively by the P4 language. This feature is briefly
+described below.
+
+## Basic principles 
+
+The C extern function can effectively enhance the functionality of P4 program. A programmer should be able to write own function, declare it in the P4 program and invoke from within P4 action or P4 control block. 
+
+A definition of the C extern function should follow the strict rules, which we call a calling convention. The calling convention describes how basic data types are passed between P4 and C, how data types are converted, how direction of P4 extern’s parameter is handled in C, etc.
+
+The C extern can use BPF helpers in order to make syscalls to eBPF subsystem. In particular, the C extern can define and have control over its own set of BPF maps. However, the C extern must not read or write to BPF maps implementing P4 tables and used by the main P4 program. 
+
+The C extern could be also allowed to access packet’s payload, but this feature is not implemented in the first version of the C Custom Externs feature.
+
+## Definition
+
+The custom C extern function should be explicitly declared in the P4 program making use of that extern. For example:
+
+```C
+/** Declaration of the C extern function. */
+extern bool verify_ipv4_checksum(in IPv4_h iphdr);
+```
+
+## Compilation
+
+The `--emit-externs` flag must be appended to the `p4c-ebpf` compiler to instruct it that 
+there are some C extern functions defined in the P4 program and compiler should not warn about them.
+
+```bash
+p4c-ebpf -o OUTPUT.c PROGRAM.p4 --emit-externs
+```
+
+Furthermore, the C file including definition of the C extern function should be provided
+to `clang`:
+
+```bash
+clang -O2 -include C-EXTERN-FILE.c -target bpf -c OUTPUT.c -o OUTPUT.o
+```
+
+## Calling convention
+
+Developer should follow the below rules while writing C extern function for P4:
```

Comment:
I don't think this line is necessary; you are just describing what the compiler does, not what the developer should do.

Reply:
Ok, I removed it.